Tests/SentryTests/Integrations/SessionReplay/SentryOnDemandReplayTests.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import Nimble
+@testable import Sentry
+import SentryTestUtils
+import XCTest
+
+#if os(iOS) || os(tvOS)
+class SentryOnDemandReplayTests: XCTestCase {
+    
+    let dateProvider = TestCurrentDateProvider()
+    let outputPath = FileManager.default.temporaryDirectory
+    
+    func getSut() -> SentryOnDemandReplay {
+        let sut = SentryOnDemandReplay(outputPath: outputPath.path, 
+                                       workingQueue: TestSentryDispatchQueueWrapper(),
```

Comment:
`m`: Thanks for using the DispatchQueueWrapper. I think we should also add one test with the actual SentryDispatchQueueWrapper to validate that our code is thread safe.

Reply:
Honestly I dont think we need.
We dont have a reliable way to test race condition, and this tests are often source of flakiness.

Reply:
you can test a lot by making the code testable. I use signals so i can control when each thread does what so the code becomes deterministic. I can find some examples when i'm on the computer 
I'd say we can merge this as add later though

Reply:
There is never a reliable way to check if your code doesn't have race conditions. For some, you need to have some test code run for minutes to surface. What I like to do is to call my code that should be thread-safe from multiple threads in a tight loop to ensure it doesn't crash right away by using all real implementations and not fakes. I did that for the metrics and it helped me to surface some threading issues https://github.com/getsentry/sentry-cocoa/blob/c02bb1f4252fce2f69a5a6e36bde70f0211663fb/Tests/SentryTests/Swift/Metrics/BucketMetricsAggregatorTests.swift#L341-L365

Ideally, we would also test if it's working correctly, but not crashing is the minimum I like to do.

Reply:
Test Added